src/ploomber/tasks/notebook.py

```diff
@@ -609,19 +626,51 @@ def __init__(
         self.check_if_kernel_installed = check_if_kernel_installed
         self.debug_mode = debug_mode
 
-        if "cwd" in self.papermill_params and self.local_execution:
-            raise KeyError(
-                'If local_execution is set to True, "cwd" should '
-                "not appear in papermill_params, as such "
-                "parameter will be set by the task itself"
+        if self.executor not in ["papermill", "ploomber-engine"]:
+            raise ValueError(
+                f"Invalid executor : {self.executor}. "
+                f"Please choose from 'papermill' or 'ploomber-engine'"
             )
 
-        if "engine" in self.papermill_params and debug_mode:
+        # We are migrating to executor_params
+        if self.papermill_params:
+            if self.executor_params == {}:
```

Comment:
cleaner way of testing if this is empty

```suggestion
            if not self.executor_params:
```